src/ts/localization/i18n.ts

```diff
@@ -0,0 +1,173 @@
+  'fullscreen': string;
+  'speed': string;
+  'playPause': string;
+}
+
+export type CustomVocabulary<V> = V & Partial<Vocabulary>;
+
+export interface BitmovinPlayerUiTranslations {
+  [key: string]: CustomVocabulary<Record<string, string>>;
+}
+
+class I18n {
+  private language: string;
+  private vocabulary: CustomVocabulary<Record<string, string>>;
+
+  constructor(config: LocalizationConfig) {
+    this.setConfig(config);
+  }
+
+  public setConfig(config: LocalizationConfig) {
+    const browserLanguageDetection = config.browserLanguageDetection != null ? config.browserLanguageDetection : true;
+    const translations = this.mergeTranslationsWithDefaultTranslations(config.translations);
+    this.initializeLanguage(config.language, browserLanguageDetection, translations);
+    this.initializeVocabulary(translations);
+  }
+
+  private containsKey(obj: object, key: string) {
+    return obj.hasOwnProperty(key);
+  }
+
+  private mergeTranslationsWithDefaultTranslations(translations: BitmovinPlayerUiTranslations = {}) {
+    const rawTranslations: BitmovinPlayerUiTranslations = { ...defaultTranslations, ...translations};
+    return Object.keys(rawTranslations).reduce((acc, key) => {
+      let translation: CustomVocabulary<Record<string, string>> = rawTranslations[key as string];
+      if (this.containsKey(defaultTranslations, key) && this.containsKey(translations, key)) {
+        translation = {...defaultTranslations[key], ...translations[key]};
+      }
+      return { ...acc, [key]: translation };
+    }, {} as BitmovinPlayerUiTranslations);
+  }
+
+  private initializeLanguage(
+    language: string,
+    browserLanguageDetectionEnabled: boolean,
+    translations: BitmovinPlayerUiTranslations,
+  ) {
+
+    if (browserLanguageDetectionEnabled) {
+      const userLanguage = (window.navigator.language);
+
+      if (translations.hasOwnProperty(userLanguage)) {
+        this.language = userLanguage;
+        return;
+      }
+      const shortenedUserLanguage = userLanguage.slice(0, 2);
+      if (translations.hasOwnProperty(shortenedUserLanguage)) {
+        this.language = shortenedUserLanguage;
+        return;
+      }
+    }
+
+    this.language = language;
+
+  }
+
+  private initializeVocabulary(translations: BitmovinPlayerUiTranslations) {
+    this.vocabulary = ['en', this.language]
+      .reduce((vocab, lang) => ({...vocab, ...(translations[lang] || {})}), {});
+  }
+
+  private replaceVariableWithPlaceholderIfExists(text: string, config: any) {
+    const matches = Array.from(text.match(new RegExp('{[a-zA-Z]+}')));
+    if (matches.length === 0) {
+      return text;
+    }
+
+    return matches
+      .map((m: string) => ({ match: m, key: m.slice(1, -1)}))
+      .reduce((str, {key, match}) => config.hasOwnProperty(key) ? str.replace(match, config[key]) : str, text);
+  }
+
+  public t<V extends CustomVocabulary<Record<string, string>> = CustomVocabulary<Record<string, string>>>(key: keyof V, config?: any) {
```

Comment:
Can `config` really be `any`? Shouldn't it be `Record<string, string | number>`?